generators/generator-base.js

```diff
@@ -101,15 +101,33 @@ module.exports = class extends PrivateBase {
      * @param {string} entityTranslationKeyMenu - i18n key for entity entry in menu
      */
     addEntityToMenu(routerName, enableTranslation, clientFramework, entityTranslationKeyMenu = _.camelCase(routerName)) {
-        if (this.clientFramework === 'angularX') {
+        if (clientFramework === 'angularX') {
             this.needleApi.clientAngular.addEntityToMenu(routerName, enableTranslation, entityTranslationKeyMenu);
-        } else if (this.clientFramework === 'react') {
+        } else if (clientFramework === 'react') {
             this.needleApi.clientReact.addEntityToMenu(routerName, enableTranslation, entityTranslationKeyMenu);
         }
     }
 
     /**
-     * Add a new entity in the TS modules file.
+     * Add a new entity in the "entities" menu.
+     *
+     * @param {string} routerName - The name of the Angular router (which by default is the name of the entity).
+     * @param {boolean} enableTranslation - If translations are enabled or not
+     * @param {string} clientFramework - The name of the client framework
+     * @param {string} entityTranslationKeyMenu - i18n key for entity entry in menu
+     */
+    addElementToAnyMenu(routerName, enableTranslation, clientFramework, entityTranslationKeyMenu, entityModule) {
```

Comment:
Refactor hint:
    - there are 5 parameters (which may be too much)
    - everytime we use the entityModule param to determine what method to call with the other 4
    - the 4 params might belong to the same object (or structure) and passing them in that order, everytime, isn't that pretty